lib/src/tasks/task_runner/api.dart

```diff
@@ -0,0 +1,118 @@
+import 'dart:async';
+import 'dart:io';
+
+import 'package:dart_dev/util.dart' show reporter, TaskProcess;
+
+import 'package:dart_dev/src/tasks/task.dart';
+
+Future<TaskRunner> runTasks(tasksToRun) async {
+  var taskGroup = new TaskGroup(tasksToRun);
+  await taskGroup.start();
+  taskGroup.output();
+
+  TaskRunner task = new TaskRunner(await taskGroup.checkExitCodes(),
+      taskGroup.taskGroupStdout, taskGroup.taskGroupStderr);
+  return task;
+}
+
+class TaskRunner extends Task {
+  final Future done = new Future.value();
+  final String stdout;
+  final String stderr;
+  bool successful;
+
+  TaskRunner(int exitCode, String this.stdout, String this.stderr) {
+    this.successful = exitCode == 0;
+  }
+}
+
+class SubTask {
+  /// Process to be executed
+  final String command;
+
+  /// A string consisting of all the subtask's output
+  String taskOutput = '';
```

Comment:
does buffering the output in a string mean we can't display output from subtasks as it happens but instead all at once once it succeeds or fails? No change requested here, just wondering.